static/skin/taskbar.js

```diff
@@ -5,24 +5,38 @@ function htmlDecode(input) {
 
 (function ($) {
     const root = $( `link[type='root']` ).attr("href");
+    const bookName = window.location.pathname.split(`${root}/`)[1].split('/')[0];
 
     $( "#kiwixsearchbox" ).autocomplete({
 
-        source: `${root}/suggest?content=${window.location.pathname.split(`${root}/`)[1].split('/')[0]}`,
+        source: `${root}/suggest?content=${bookName}`,
         dataType: "json",
         cache: false,
 
         response: function( event, ui ) {
 
               for(const item of ui.content) {
                   item.label = htmlDecode(item.label);
-                  item.value = htmlDecode(item.value);
+                  item.value = encodeURI(htmlDecode(item.value));
+
+                  if (item.label === `containing '${item.value}'...`) {
+                      item.value = `${root}/search?content=${bookName}&pattern=${item.value}+`;
```

Comment:
(This is a open questioning, not a asked change)

The `+`(space) at the end of the request is to force a real search (avoid a redirection in the backend if the title exact matches https://github.com/kiwix/kiwix-lib/blob/master/src/server/internalServer.cpp#L514-L519).

If we remove the redirection in the backend, we don't need to a space at the end of the query.
As for suggestion, it may break some third party user (I don't know about any).
@kelson42 may have a opinion about this but I think we can remove this redirection in the backend and make the `search` endpoint always do a search.